### PR TITLE
added new jest tests for sidebar and for data selector

### PR DESCRIPTION
**Pull request by sumukhswamy**
### Description
Jest tests for Sidebar which checks Databases, tables, skipping index, covering index, materialized view
Jest tests for data selector
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .cypress/integration/ui.spec.js [hunk 1]
```diff
@@ -37,32 +37,34 @@ describe('Test PPL UI', () => {
...
-  it('Confirm results are empty', () => {
-    cy.get('.euiTextAlign')
-      .contains('Enter a query in the query editor above to see results.')
-      .should('have.length', 1);
-  });
+  // it('Confirm results are empty', () => {
+  //   cy.get('.euiTextAlign')
+  //     .contains('Enter a query in the query editor above to see results.')
+  //     .should('have.length', 1);
+  // });
```
Comment (ps48) [inline .cypress/integration/ui.spec.js:48]: We can remove the comments

File: .cypress/integration/ui.spec.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ describe('Dump test data', () => {
       cy.request(url).then((response) => {
         cy.request({
           method: 'POST',
-          form: true,
+          form: false,
```
Comment (ps48) [inline .cypress/integration/ui.spec.js:18]: Are cypress changes intended in this PR.

File: public/components/Main/main.test.tsx [hunk 3]
```diff
@@ -32,6 +34,76 @@ describe('<Main /> spec', () => {
...
+    const pplButton = getByText('PPL');
+    const asyncTest = () => {
+      fireEvent.click(pplButton);
+    };
+    waitFor(()=>{
+      asyncTest();
+    })
+    expect(document.body.children[0]).toMatchSnapshot();
+  });
+  it('renders the component and checks if Opensearch is selected', async () => {
```
Comment (ps48) [inline public/components/Main/main.test.tsx:53]: Can you please run the formatter here?

File: .cypress/integration/ui.spec.js [hunk 4]
```diff
@@ -231,17 +218,15 @@ describe('Test table display', () => {
...
       .focus()
-      .type(`{selectall}{backspace}select * from employee_nested;`, { force: true });
-    cy.wait(delay);
+      .type(`{selectall}{backspace}select * from employee_nested;`, {
+        force: true,
+      });
+    cy.wait(1000);
     cy.get('.euiButton__text').contains('Run').click({ force: true });
-    cy.wait(delay);
+    cy.wait(1000);
```
Comment (ps48) [inline .cypress/integration/ui.spec.js:226]: We were refraining from using cypress wait. Did you try it without the wait being added here?

File: public/components/SQLPage/table_view.tsx [hunk 5]
```diff
@@ -216,6 +216,7 @@ export const TableView = ({ http, selectedItems, updateSQLQueries, refreshTree }
         setToast(errorMessage, 'danger');
       } else {
         pollQueryStatus(id, http, (data) => {
+          console.log(data,'here')
```
Comment (ps48) [inline public/components/SQLPage/table_view.tsx:219]: Let's remove all console logs and commented code.

File: public/components/SQLPage/__tests__/table_view.test.tsx [hunk 6]
```diff
@@ -5,57 +5,310 @@
...
           updateSQLQueries={() => {}}
           refreshTree={false}
         />
       );
     };
     await asyncTest();
+    expect(screen.getByLabelText('Sample Folder Tree')).toBeInTheDocument();
+    expect(document.body.children[0]).toMatchSnapshot();
+  });
+  it('fetches default databse in the side tree', async () => {
```
Comment (ps48) [inline public/components/SQLPage/__tests__/table_view.test.tsx:73]: `database` typo?

File: public/components/SQLPage/__tests__/DataSelect.test.tsx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    wrapper.update();
+    await waitFor(() => {
+      expect(
+        toJson(wrapper, {
+          noKey: false,
+          mode: 'deep',
+        })
+      ).toMatchSnapshot();
+    });
+    // expect(screen.getByText('OpenSearch')).toBeInTheDocument();
```
Comment (ps48) [inline public/components/SQLPage/__tests__/DataSelect.test.tsx:68]: can remove the comment

File: .cypress/integration/ui.spec.js (hunk 2 above)
Comment (sumukhswamy) [reply .cypress/integration/ui.spec.js:18]: FTR repo also has this as false, do I still revert it?

File: .cypress/integration/ui.spec.js (hunk 4 above)
Comment (sumukhswamy) [reply .cypress/integration/ui.spec.js:226]: without wait its failing, I have reverted to waht it was originally which is cy.wait(delay);

Comment (ps48) [reply .cypress/integration/ui.spec.js]: We need to find a way around using wait. Can you try changing the test to use should. You can take a look at changes done here: https://github.com/opensearch-project/dashboards-visualizations/blob/d1551b78fdca45c376e82a5a32374fd61ccd03d4/gantt-chart/.cypress/integration/ui.spec.js#L132-L142
Original PR to remove wait: https://github.com/opensearch-project/dashboards-visualizations/pull/314/files

File: .cypress/integration/ui.spec.js (hunk 2 above)
Comment (ps48) [reply .cypress/integration/ui.spec.js:18]: With the comment I wanted to know if cypress updates are part of this PR, cause the PR description only talks about jest updates.
```
Jest tests for Sidebar which checks Databases, tables, skipping index, covering index, materialized view
Jest tests for data selector

**Pull request closed without merging**